fix: overflow of link and warnings

montelaidev opened the pull request:
## Explanation

Changes:
- Creates a new snap detail label.
- Fixes the snap url overflowing. 
- Fixes console errors. 

Fixes [#29](https://github.com/MetaMask/accounts-planning/issues/29)
-->

## Screenshots/Screencaps

### Before
![fix-29-snap-url-overflow-before](https://github.com/MetaMask/metamask-extension/assets/96463427/c96b01d7-9239-4089-84ea-1ca205292f69)

### After examples
(Left expanded and right reduced)
<div style="display: inline-block"><img width="400" src="https://github.com/MetaMask/metamask-extension/assets/32695229/c545e790-42b2-40a7-b8a1-ac17d9bd682e"><img width="129" src="https://github.com/MetaMask/metamask-extension/assets/32695229/c8d199c7-9a09-441b-8bf6-c9547d9db123"</div>


## Manual Testing Steps

- Click on the Account Menu
- Click on Add snap account
- Click on any snap listed and go to the details. 

## Pre-merge author checklist

- [ x] I've clearly explained:
  - [ x] What problem this PR is solving
  - [ x] How this problem was solved
  - [ x] How reviewers can test my changes
- [ x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [x ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ x] PR is linked to the appropriate GitHub issue

montelaidev committed:
fix: overflow of link and warnings

montelaidev committed:
Merge branch 'develop' into fix-29-snap-url-text-overflow

montelaidev committed:
Merge remote-tracking branch 'origin/develop' into fix-29-snap-url-text-overflow

montelaidev committed:
fix: update snapshot

gantunesr reviewed (dismissed):
LGTM. Just a small suggestion

HowardBraham committed:
Merge branch 'develop' into fix-29-snap-url-text-overflow

HowardBraham committed:
--wrapped in React.memo HOC
--changed Website to a clickable link
--slightly better overflow wrapping, but the best would be to wrap on dots and slashes

HowardBraham committed:
updated snapshot

plasmacorral commented:
We have a conflict that will need to be resolved.

HowardBraham committed:
Merge branch 'develop' into fix-29-snap-url-text-overflow

HowardBraham committed:
Merge branch 'develop' into fix-29-snap-url-text-overflow
